src/app/_components/agencycomponents/agency-interactive-area.tsx

```diff
@@ -23,12 +25,22 @@ const GOOGLE_MAPS_LIBRARIES_ARRAY: Libraries = ["places"]; //Add more to this ar
 export const BookingInteractiveArea = ({ initialViewMode = ViewMode.CALENDAR }: Props) => {
   const [viewMode, setViewMode] = useState<ViewMode>(initialViewMode);
   const [showBookingModal, setShowBookingModal] = useState<boolean>(false);
-  // eventually this loading state will be replacted with a tanstack mutation loading state
-  const [loading, setLoading] = useState<boolean>(false);
   const [currentDate, setCurrentDate] = useState<Date>(new Date());
   const [isDayView, setIsDayView] = useState<boolean>(false);
   const [validationAddressGood, setValidationAddressGood] = useState<boolean>(false);
 
+  const createBookingMutation = api.trip.create.useMutation({
+    onSuccess: () => {
+      notify.success("Booking successfully created");
+      form.reset();
+      setShowBookingModal(false);
+      setValidationAddressGood(false);
```

Comment:
after creating a new booking, the table and calendar wont refresh. perhaps we should invalidate the bookings query to cause a refresh
```suggestion
      setValidationAddressGood(false);
      void utils.bookings.invalidate();
```